src/lua/arrow_record_batch.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2024, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "arrow_record_batch.h"
+
+#include "lua/utils.h"
+
+static const char *arrow_record_batch_typename = "arrow_record_batch";
+
+struct arrow_record_batch *
+luaT_new_arrow_record_batch(struct lua_State *L)
+{
+	struct arrow_record_batch *arrow = lua_newuserdata(L, sizeof(*arrow));
+	memset(arrow, 0, sizeof(*arrow));
+	luaL_getmetatable(L, arrow_record_batch_typename);
+	lua_setmetatable(L, -2);
+	return arrow;
+}
+
+struct arrow_record_batch *
+luaT_check_arrow_record_batch(struct lua_State *L, int idx)
+{
+	return luaL_checkudata(L, idx, arrow_record_batch_typename);
```

Comment:
Let's use `luaT_checkudata` here as we are going to use box errors instead of Lua. Then we don't need `t.assert_error_msg_content_equals` in the tests.

Reply:
Done.